// @unite-us/ui/styleguide.config.js
/* eslint-disable */
const path = require('path');

module.exports = {
  title: 'Unite Us UI components',
  usageMode: 'expand',
  webpackConfig: require('./config/webpack.config.dev.js'),
  styleguideComponents: {
    Wrapper: path.join(__dirname, 'src/Wrapper')
  },
  skipComponentsWithoutExample: true,
  require: [
    path.resolve(__dirname, 'config/styleguide.setup.js')
  ],
  styles: {
    StyleGuide: {
      '@global body': {
        fontFamily: 'Avenir Roman',
      }
    }
  },
  sections: [
    {
      name: 'Typography',
      content: './src/typography.md'
    },
    {
      name: 'Display Elements',
      components: () => [
        './src/components/Display/Avatar/index.jsx',
        './src/components/Display/Badge/index.jsx',
        './src/components/Display/Button/index.jsx',
        './src/components/Display/Divider/index.jsx',
        './src/components/Display/Drawer/index.jsx',
        './src/components/Display/Expandable/index.jsx',
        './src/components/Display/Gradient/index.jsx',
        './src/components/Display/Icon/index.jsx',
        './src/components/Display/Menu/index.jsx',
        './src/components/Display/MoreOrLess/index.jsx',
        './src/components/Display/Pager/index.jsx',
        './src/components/Display/Popover/index.jsx',
        './src/components/Display/SelectionBubble/index.jsx',
      ],
      sections: [
        {
          name: 'AppBar',
          components: () => [
            './src/components/Display/AppBar/index.jsx',
            './src/components/Display/AppBar/components/AppSideNav.jsx',
          ]
        },
        {
          name: 'BaseCard',
          components: () => [
            './src/components/Display/BaseCard/index.jsx',
            './src/components/Display/BaseCard/components/BaseCardHeader.jsx',
            './src/components/Display/BaseCard/components/BaseCardBody.jsx',
          ]
        },
        {
          name: 'Card',
          components: () => [
            './src/components/Display/Card/index.jsx',
            './src/components/Display/Card/components/CardHeader.jsx',
            './src/components/Display/Card/components/CardText.jsx',
            './src/components/Display/Card/components/CardActions.jsx',
          ]
        },
        {
          name: 'Tables',
          components: () => [
            './src/components/Display/Table/index.jsx',
            './src/components/Display/Table/components/TableTitle.jsx',
            './src/components/Display/Table/components/TableRow.jsx',
            './src/components/Display/Table/components/TableRowColumn.jsx',
            './src/components/Display/Table/components/TableHeader.jsx',
            './src/components/Display/Table/components/TableHeaderColumn.jsx',
            './src/components/Display/Table/components/TableBody.jsx',
          ]
        },
        {
          name: 'Loader',
          components: () => [
            './src/components/Display/Loader/BarLoader.jsx',
            './src/components/Display/Loader/IconLoader.jsx',
          ]
        },
        {
          name: 'Tabs',
          components: () => [
            './src/components/Display/Tabs/index.jsx',
            './src/components/Display/Tabs/components/Tab.jsx',
          ]
        },
        {
          name: 'Menu',
          components: () => [
            './src/components/Display/Menu/index.jsx',
            './src/components/Display/Menu/components/MenuItem.jsx',
          ]
        }
      ]
    },
    {
      name: 'Form Elements',
      content: './src/components/Form/utils/fieldProps.md',
      components: './src/components/Form/**/*.jsx',
    },
    {
      name: 'Composed Elements',
      components: './src/components/Composed/**/*.jsx',
    },
    {
      name: 'UniteUS colors',
      components: './src/components/Colors/index.jsx',
    },
  ],
};
