// @unite-us/ui/dist/Form/RadioField/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = _interopRequireDefault(require("lodash"));

var _classnames = _interopRequireDefault(require("classnames"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); Object.defineProperty(subClass, "prototype", { writable: false }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

/**
 * RadioField Component
 *
*/
var RadioField = /*#__PURE__*/function (_Component) {
  _inherits(RadioField, _Component);

  var _super = _createSuper(RadioField);

  function RadioField(props) {
    var _this;

    _classCallCheck(this, RadioField);

    _this = _super.call(this, props);
    _this.onChange = _this.onChange.bind(_assertThisInitialized(_this));
    _this.state = {
      loading: false
    };
    return _this;
  }

  _createClass(RadioField, [{
    key: "onChange",
    value: function onChange(value) {
      var callback = (0, _utils.getPreferredProp)(this.props, 'onChange');

      if (_lodash["default"].isFunction(callback)) {
        callback(value);
      }

      if (_lodash["default"].isFunction(this.props.onChange) && !_lodash["default"].isEqual(this.props.onChange, (0, _utils.getPreferredProp)(this.props, 'onChange'))) {
        this.props.onChange(value);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          options = _this$props.options,
          className = _this$props.className,
          labelClassName = _this$props.labelClassName,
          id = _this$props.id,
          style = _this$props.style,
          inline = _this$props.inline,
          label = _this$props.label,
          inlineLabel = _this$props.inlineLabel,
          postLabelContent = _this$props.postLabelContent,
          required = _this$props.required;
      var hasError = (0, _utils.getPreferredProp)(this.props, 'invalid') && (0, _utils.getPreferredProp)(this.props, 'touched');

      var containerClass = function containerClass() {
        return (0, _classnames["default"])({
          'ui-radio-field': true,
          'ui-radio-field--inline': inline,
          'ui-form-field': true,
          'ui-form-field--has-error': hasError,
          'ui-radio-field--has-error': hasError
        }, className);
      };

      var labelClass = function labelClass() {
        return (0, _classnames["default"])({
          'ui-form-field__label': true,
          'ui-form-field__label--left': inlineLabel,
          'ui-radio-field__label': true,
          'ui-form-field__label--required': required,
          'ui-form-field__label--roman-label': !_lodash["default"].isEmpty(labelClassName) && labelClassName === 'ui-label-roman ui-label-roman--lowercase'
        });
      };

      return /*#__PURE__*/_react["default"].createElement("div", {
        className: containerClass(),
        style: style.container
      }, !_lodash["default"].isEmpty(label) && /*#__PURE__*/_react["default"].createElement("div", {
        className: labelClass(),
        style: style.label
      }, label), postLabelContent, _lodash["default"].map(options, function (opt, index) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          className: "ui-radio-field__item",
          key: index
        }, /*#__PURE__*/_react["default"].createElement("label", {
          htmlFor: "".concat(id, "-").concat(index)
        }, /*#__PURE__*/_react["default"].createElement("input", {
          id: "".concat(id, "-").concat(index),
          type: "radio",
          name: (0, _utils.getPreferredProp)(_this2.props, 'name'),
          checked: (0, _utils.getPreferredProp)(_this2.props, 'value') === opt.value,
          onChange: _this2.onChange,
          onBlur: (0, _utils.getPreferredProp)(_this2.props, 'onBlur'),
          onFocus: (0, _utils.getPreferredProp)(_this2.props, 'onFocus'),
          disabled: opt.disabled,
          style: style.input,
          value: opt.value
        }), /*#__PURE__*/_react["default"].createElement("span", {
          id: "".concat(opt.value, "-label")
        }, opt.label)));
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: "ui-form-field__error",
        style: style.error
      }, (0, _utils.getPreferredProp)(this.props, 'error')));
    }
  }]);

  return RadioField;
}(_react.Component);

RadioField.propTypes = _objectSpread({
  /** id for reference */
  id: _propTypes["default"].string.isRequired,

  /** className for reference */
  className: _propTypes["default"].string,

  /** array of options consist of value and label */
  options: _propTypes["default"].array,

  /** Text of the label */
  label: _propTypes["default"].string,

  /** If true, Radio Buttons will be displayed inline */
  inline: _propTypes["default"].bool,

  /** If true, Radio Label will be displayed inline */
  inlineLabel: _propTypes["default"].bool,

  /** Content block that is placed between the label and the field */
  postLabelContent: _propTypes["default"].node,

  /** Is the field required */
  required: _propTypes["default"].bool,

  /** Override the inline-styles of the elements || **Not Recommended** */
  style: _propTypes["default"].shape({
    container: _propTypes["default"].object,
    label: _propTypes["default"].object,
    text: _propTypes["default"].object,
    error: _propTypes["default"].object
  })
}, _utils.fieldProps);
RadioField.defaultProps = {
  id: '',
  className: '',
  options: [],
  inlineLabel: false,
  label: '',
  labelClassName: '',
  postLabelContent: null,
  required: false,
  style: {
    container: {},
    label: {},
    input: {},
    error: {}
  },
  inline: false
};
var _default = RadioField;
exports["default"] = _default;
