// @unite-us/ui/dist/Display/Card/components/CardActions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _filterChildren = _interopRequireDefault(require("../../../utils/Children/filterChildren"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var CardActions = function CardActions(props) {
  var id = props.id,
      className = props.className,
      expanded = props.expanded,
      expandable = props.expandable,
      style = props.style;

  var cardActionClass = function cardActionClass() {
    return (0, _classnames["default"])({
      'ui-card-actions': true,
      'ui-card-actions--hidden': !expanded && expandable,
      'ui-card-actions--expanded': expanded || !expandable
    }, className);
  };

  var newChildren = (0, _filterChildren["default"])(props.children);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: cardActionClass(),
    id: id,
    style: style
  }, newChildren);
};

CardActions.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for reference */
  className: _propTypes["default"].string,

  /** Override the inline-styles of the root element || Not Recommended */

  /**
   * If true, this card component is controlled.
   * @ignore
   */
  expandable: _propTypes["default"].bool,

  /**
   * If true, this card component is expanded (uncontrolled)
   * @ignore
   */
  expanded: _propTypes["default"].bool,
  style: _propTypes["default"].object,

  /**
   * children props for content inside
   * @ignore
   */
  children: _propTypes["default"].node
};
CardActions.defaultProps = {
  id: '',
  className: '',
  style: {},
  children: [],
  expanded: null,
  expandable: false
};
var _default = CardActions;
exports["default"] = _default;
