// @unite-us/ui/dist/Display/AppBar/components/AppSideNav.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _filterChildren = _interopRequireDefault(require("../../../utils/Children/filterChildren"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var AppSideNav = function AppSideNav(props) {
  var id = props.id,
      className = props.className,
      children = props.children,
      style = props.style,
      width = props.width,
      navContent = props.navContent,
      containerStyle = props.containerStyle,
      mainContentStyle = props.mainContentStyle,
      opened = props.opened,
      containerClassName = props.containerClassName;

  var appSideNavContainerClass = function appSideNavContainerClass() {
    return (0, _classnames["default"])({
      'ui-app-bar-side-nav': true,
      'ui-app-bar-side-nav--opened': opened
    }, containerClassName);
  };

  var appBarSideNavClass = function appBarSideNavClass() {
    return (0, _classnames["default"])({
      'ui-app-bar-side-nav__side-content': true
    }, className);
  };

  var appBarSideNavMainClass = function appBarSideNavMainClass() {
    return (0, _classnames["default"])({
      'ui-app-bar-side-nav__main-content': true
    });
  };

  var newChildren = (0, _filterChildren["default"])(children);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: appSideNavContainerClass(),
    style: _objectSpread(_objectSpread({}, containerStyle), {}, {
      width: opened ? '100%' : "calc(100% + ".concat(width, ")"),
      transform: opened ? 'translateX(0)' : "translateX(-".concat(width, ")")
    })
  }, /*#__PURE__*/_react["default"].createElement("div", {
    id: id,
    className: appBarSideNavClass(),
    style: style
  }, navContent), /*#__PURE__*/_react["default"].createElement("div", {
    className: appBarSideNavMainClass(),
    style: mainContentStyle
  }, newChildren));
};

AppSideNav.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for sidebar nav */
  className: _propTypes["default"].string,

  /** styles for the sidebar nav */
  style: _propTypes["default"].object,

  /** className for container of the side nav */
  containerClassName: _propTypes["default"].string,

  /** styles for container of the side nav */
  containerStyle: _propTypes["default"].object,

  /** styles for the main content */
  mainContentStyle: _propTypes["default"].object,

  /** width of the side bar nav */
  width: _propTypes["default"].string,

  /** content inside of the nav side bar */
  navContent: _propTypes["default"].node.isRequired,

  /** if true, nav side bar will show */
  opened: _propTypes["default"].bool,

  /**
  * children props for content inside
  * @ignore
  */
  children: _propTypes["default"].node
};
AppSideNav.defaultProps = {
  id: '',
  className: '',
  containerClassName: '',
  containerStyle: {},
  children: [],
  style: {},
  navContent: null,
  mainContentStyle: {},
  opened: false,
  width: '250px'
};
var _default = AppSideNav;
exports["default"] = _default;
