// @unite-us/ui/dist/Display/Table/components/TableHeaderColumn.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _lodash = _interopRequireDefault(require("lodash"));

var _filterChildren = _interopRequireDefault(require("../../../utils/Children/filterChildren"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var TableHeaderColumn = function TableHeaderColumn(props) {
  var className = props.className,
      id = props.id,
      style = props.style,
      secondaryDataList = props.secondaryDataList,
      columnIndex = props.columnIndex,
      children = props.children;

  var tableHeaderColumnClass = function tableHeaderColumnClass() {
    return (0, _classnames["default"])({
      'ui-table-header-column': true,
      'hidden-sm-down': _lodash["default"].includes(secondaryDataList, columnIndex)
    }, className);
  };

  var newChildren = (0, _filterChildren["default"])(children);
  return /*#__PURE__*/_react["default"].createElement("th", {
    id: id,
    className: tableHeaderColumnClass(),
    style: style
  }, newChildren);
};

TableHeaderColumn.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for reference */
  className: _propTypes["default"].string,

  /** Override the inline-styles of the root element || Not Recommended */
  style: _propTypes["default"].object,

  /**
   * children props for content inside
   * @ignore
   */
  children: _propTypes["default"].node,

  /**
   * Column index passed from TableRow component
   * @ignore
   */
  columnIndex: _propTypes["default"].number.isRequired,

  /**
   * Data (in columns) to hide passed from TableRow component
   * @ignore
   */
  secondaryDataList: _propTypes["default"].array
};
TableHeaderColumn.defaultProps = {
  id: '',
  className: '',
  children: [],
  style: {},
  secondaryDataList: [],
  columnIndex: 0
};
var _default = TableHeaderColumn;
exports["default"] = _default;
