// @unite-us/ui/dist/Display/Table/components/TableBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _filterChildren = _interopRequireDefault(require("../../../utils/Children/filterChildren"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var TableBody = function TableBody(props) {
  var className = props.className,
      id = props.id,
      children = props.children,
      style = props.style,
      secondaryDataList = props.secondaryDataList;

  var tableBodyClass = function tableBodyClass() {
    return (0, _classnames["default"])({
      'ui-table-body': true
    }, className);
  };

  var filteredChildren = (0, _filterChildren["default"])(children);
  var childrenWithFuncs;

  if (props.onCellClick) {
    childrenWithFuncs = _react["default"].Children.map(filteredChildren, function (child, index) {
      return /*#__PURE__*/_react["default"].cloneElement(child, {
        onCellClick: props.onCellClick,
        parent: 'table-body',
        rowIndex: index,
        secondaryDataList: secondaryDataList
      });
    });
  } else {
    childrenWithFuncs = _react["default"].Children.map(filteredChildren, function (child, index) {
      return /*#__PURE__*/_react["default"].cloneElement(child, {
        parent: 'table-body',
        rowIndex: index,
        secondaryDataList: secondaryDataList
      });
    });
  }

  return /*#__PURE__*/_react["default"].createElement("tbody", {
    id: id,
    className: tableBodyClass(),
    style: style
  }, childrenWithFuncs);
};

TableBody.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for reference */
  className: _propTypes["default"].string,

  /** prop fucntion when user clicks */
  onCellClick: _propTypes["default"].func,

  /** Override the inline-styles of the root element || Not Recommended */
  style: _propTypes["default"].object,

  /**
   * children props for content inside
   * @ignore
   */
  children: _propTypes["default"].node,

  /**
   * on cell click passed from the Table component
   * @ignore
   */

  /**
   * Data (in columns) to hide passed from TableRow component
   * @ignore
   */
  secondaryDataList: _propTypes["default"].array
};
TableBody.defaultProps = {
  id: '',
  className: '',
  children: [],
  style: {},
  onCellClick: null,
  secondaryDataList: []
};
var _default = TableBody;
exports["default"] = _default;
