// @unite-us/ui/dist/Display/Loader/BarLoader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _lodash = _interopRequireDefault(require("lodash"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var BarLoader = function BarLoader(props) {
  var tall = props.tall,
      className = props.className,
      id = props.id,
      size = props.size,
      style = props.style;

  var barLoaderClass = function barLoaderClass() {
    return (0, _classnames["default"])({
      'bar-loader': true,
      'bar-loader--tall': tall,
      'bar-loader--quarter': _lodash["default"].isEqual('quarter', size),
      'bar-loader--semi-full': _lodash["default"].isEqual('semi-full', size),
      'bar-loader--half': _lodash["default"].isEqual('half', size)
    }, className);
  };

  return /*#__PURE__*/_react["default"].createElement("div", {
    id: id,
    style: style,
    className: barLoaderClass()
  });
};

BarLoader.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for reference */
  className: _propTypes["default"].string,

  /** determine if you want to use taller height */
  tall: _propTypes["default"].bool,

  /** width size of loader bar */
  size: _propTypes["default"].oneOf(['quarter', 'half', 'semi-full', 'full']),

  /** Override the inline-styles of the root element || Not Recommended */
  style: _propTypes["default"].object
};
BarLoader.defaultProps = {
  id: '',
  className: '',
  tall: false,
  size: 'full',
  style: {}
};
var _default = BarLoader;
exports["default"] = _default;
