// @unite-us/ui/dist/Display/Hamburger/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _lodash = require("lodash");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Hamburger = function Hamburger(props) {
  var opened = props.opened,
      id = props.id,
      className = props.className,
      style = props.style;

  var hamburgerContainerClass = function hamburgerContainerClass() {
    return (0, _classnames["default"])({
      'ui-hamburger': true
    }, className);
  };

  var bar1Class = function bar1Class() {
    return (0, _classnames["default"])({
      'ui-hamburger__bar1': true,
      'ui-hamburger__bar1--opened': opened
    });
  };

  var bar2Class = function bar2Class() {
    return (0, _classnames["default"])({
      'ui-hamburger__bar2': true,
      'ui-hamburger__bar2--opened': opened
    });
  };

  var bar3Class = function bar3Class() {
    return (0, _classnames["default"])({
      'ui-hamburger__bar3': true,
      'ui-hamburger__bar3--opened': opened
    });
  };

  return /*#__PURE__*/_react["default"].createElement("div", {
    id: id,
    className: hamburgerContainerClass(),
    onClick: props.onMenuClick,
    role: "button",
    tabIndex: 0,
    style: style
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: bar1Class()
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: bar2Class()
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: bar3Class()
  }));
};

Hamburger.propTypes = {
  /** id for reference */
  id: _propTypes["default"].string,

  /** className for reference */
  className: _propTypes["default"].string,

  /** Override the inline-styles of the root element || Not Recommended */
  style: _propTypes["default"].object,

  /** If true, this card component is expandable. */
  opened: _propTypes["default"].bool.isRequired,

  /** Callback function when the burger icon is clicked */
  onMenuClick: _propTypes["default"].func
};
Hamburger.defaultProps = {
  id: '',
  className: '',
  style: {},
  onMenuClick: _lodash.noop
};
var _default = Hamburger;
exports["default"] = _default;
