// @unite-us/ui/dist/Colors/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/* eslint-disable react/no-array-index-key */
var COLORS = {
  main: {
    name: 'Main',
    colors: [{
      className: 'display-color',
      type: 'display color',
      variable: '$display-color',
      code: '#2C405A'
    }, {
      className: 'body-grey-color',
      type: 'body grey color',
      variable: '$body-grey-color',
      code: '#506784'
    }, {
      className: 'active-color',
      type: 'active color',
      variable: '$active-color',
      code: '#67859E'
    }, {
      className: 'field-border-color',
      type: 'field border color',
      variable: '$field-border-color',
      code: '#D8E5EE'
    }, {
      className: 'cloud-blue-color',
      type: 'cloud blue color',
      variable: '$cloud-blue-color',
      code: '#ECF5FD'
    }, {
      className: 'table-rule-color',
      type: 'table ruler color',
      variable: '$table-rule-color',
      code: '#E5EEF5'
    }, {
      className: 'table-border-color',
      type: 'table border color',
      variable: '$table-border-color',
      code: '#B7D2E5(.8)'
    }, {
      className: '$section-color',
      type: 'section border color',
      variable: '$section-color',
      code: '#FAFBFC'
    }]
  },
  active_colors: {
    name: 'Active Colors',
    colors: [{
      className: 'united-blue-color',
      type: 'united blue color',
      variable: '$united-blue-color',
      code: '#4571BA'
    }, {
      className: 'sky-blue-color',
      type: 'sky blue color',
      variable: 'sky-blue-color',
      code: '#628ED7'
    }, {
      className: 'deep-blue-color',
      type: 'deep blue color',
      variable: 'deep-blue-color',
      code: '#2D5495'
    }, {
      className: 'grass-color',
      type: 'grass color',
      variable: '$grass-color',
      code: '#4ECE3D'
    }, {
      className: 'eggplant-color',
      type: 'eggplant color',
      variable: '$eggplant-color',
      code: '#712F79'
    }]
  },
  status: {
    name: 'Status Colors',
    colors: [{
      className: 'canary-color',
      type: 'canary color',
      variable: '$canary-color',
      code: '#F4D316'
    }, {
      className: 'tangie-color',
      type: 'tangie color',
      variable: '$tangie-color',
      code: '#FF9900'
    }, {
      className: 'rosso-color',
      type: 'rosso color',
      variable: '$rosso-color',
      code: '#D0021B'
    }, {
      className: 'united-blue-color',
      type: 'united blue color',
      variable: '$united-blue-color',
      code: '#4571BA'
    }, {
      className: 'grass-color',
      type: 'grass color',
      variable: '$grass-color',
      code: '#4ECE3D'
    }, {
      className: 'avocado-color',
      type: 'avocado color',
      variable: 'avocado-color',
      code: '#078b00'
    }, {
      className: 'silver-color',
      type: 'silver color',
      variable: '$silver-color',
      code: '#CACACA'
    }, {
      className: 'lil-grey-color',
      type: 'lil grey color',
      variable: 'lil-grey-color',
      code: '#F9F9F9'
    }]
  }
};

function mapColorDivs(color) {
  return COLORS[color].colors.map(function (c, i) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "col-xs-12 col-sm-2 colors__grid",
      key: "color-".concat(i)
    }, /*#__PURE__*/_react["default"].createElement("section", {
      className: "colors__section"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "colors__palette-container color-background--".concat(c.className)
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "colors__details"
    }, /*#__PURE__*/_react["default"].createElement("small", {
      className: "colors__type"
    }, c.type), /*#__PURE__*/_react["default"].createElement("small", {
      className: "colors__variable"
    }, c.variable), /*#__PURE__*/_react["default"].createElement("span", null, /*#__PURE__*/_react["default"].createElement("strong", {
      className: "uppercase"
    }, c.code)))));
  });
}

var Colors = function Colors() {
  var mainDivs = mapColorDivs('main');
  var activeDivs = mapColorDivs('active_colors');
  var statusDivs = mapColorDivs('status');
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "colors"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "row colors__row"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "col-xs-12"
  }, /*#__PURE__*/_react["default"].createElement("h4", null, COLORS.main.name)), mainDivs), /*#__PURE__*/_react["default"].createElement("div", {
    className: "row colors__row"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "col-xs-12"
  }, /*#__PURE__*/_react["default"].createElement("h4", null, COLORS.active_colors.name)), activeDivs), /*#__PURE__*/_react["default"].createElement("div", {
    className: "row colors__row"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "col-xs-12"
  }, /*#__PURE__*/_react["default"].createElement("h4", null, COLORS.status.name)), statusDivs));
};

var _default = Colors;
exports["default"] = _default;
